torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -878,6 +878,19 @@ RegisterOperators reg(
              return 0;
            };
          }),
+     Operator(
+         "aten::select0(int[] self) -> int",
```

Comment:
Instead of creating this new op, could you instead compose it of our existing ops? 
if self.dim() == 0:
      raise TypeError('iteration over a 0-d tensor')
select_index = self.size(0)

Reply:
Hmmm, I don't know. I think this will make IR more convoluted and less readable. Unless we have optimizations that can automatically infer this type of constraints, propagate and benefit from it, I am not sure if all this extra control flow is worth it.

Reply:
We specialize on dimensions - so we do have these optimizations

Reply:
I don't think we should be optimizing for IR readability in its unoptimized state, and I think in the long run adding more and more ad-hoc Ops will make our IR less readable

Reply:
> We specialize on dimensions - so we do have these optimizations

Could you please point me to the code that records the dimensionality of list types or any more general constraints on it?
I'd like to understand this better.

Reply:
Sure, look at peephole.cpp the line: node->matches("aten::dim(Tensor self) -> int. Then in constant propagation we remove if statements which have a False conditional. So the error checking here will be optimized away.

Reply:
Looking into it we already have a `aten::select` that does the right thing from ATen, its signature is `aten::select(Tensor(a) self, int dim, int index)`, so there's no need for a new op

Reply:
Switched to `aten::dim`. Thanks to @eellison for the suggestion. I didn't realize it was defined on a tensor rather than list.

Reply:
@driazati please take another look!